ProcessOptimizer/optimizer/optimizer.py

```diff
@@ -467,11 +467,13 @@ def ask(self, n_points=None, strategy="stbr_fill"):
                 + "got %s" % strategy
             )
         
-        if strategy in ["stbr_fill", "stbr_full"] and self.get_constraints() is not None:
-            raise ValueError(
-                "Steinerberger (default setting) sampling can not be used with constraints,\
-                try using another strategy like 'opt.ask(n,strategy='cl_min')'"
-            )
+        if (n_points > self._n_initial_points 
+            and strategy in ["stbr_fill", "stbr_full"] 
+            and self.get_constraints() is not None):
+                raise ValueError(
```

Comment:
We could also make this a bespoke error class, then you could catch that specific error in your script.

Reply:
But then you still need the points, which you won't get...